modules/configuration/php/diagnosis.class.inc

```diff
@@ -0,0 +1,288 @@
+     */
+    private function _handleGET() : ResponseInterface
+    {
+        $response = $this->initializeSetup();
+        return new \LORIS\Http\Response\JsonResponse(
+            $response
+        );
+    }
+
+    /**
+     * Loads the project page.
+     *
+     * @return array
+     */
+    function initializeSetup() : array
+    {
+        $DB = \NDB_Factory::singleton()->database();
+
+        $projects    = \Utility::getProjectList();
+        $visits      = \Utility::getVisitList();
+        $instruments = \Utility::getAllInstruments();
+
+        $parameterType = $DB->pselect(
+            "SELECT DISTINCT SourceField FROM parameter_type
+            WHERE Queryable='1'
+            ORDER BY SourceField",
+            []
+        );
+
+        $sourceFields = [];
+        foreach ($parameterType as $record) {
+            $sourceFields[$record['SourceField']] = $record['SourceField'];
+        }
+
+        $diagnosisTracks = $DB->pselectWithIndexKey(
+            "SELECT * FROM diagnosis_evolution
+            ORDER BY orderNumber",
+            [],
+            'DxEvolutionID'
+        );
+
+        foreach ($diagnosisTracks as $key => $data) {
+            $diagnosisTracks[$key]['sourceField']
+                = explode(",", $diagnosisTracks[$key]['sourceField']);
+        }
+
+        $values['projects']        = $projects;
+        $values['visits']          = $visits;
+        $values['instruments']     = $instruments;
+        $values['sourceFields']    = $sourceFields;
+        $values['diagnosisTracks'] = $diagnosisTracks;
+
+        return $values;
+    }
+
+    /**
+     * Processes the values & saves to database and return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePOST(ServerRequestInterface $request) : ResponseInterface
+    {
+        $values = $request->getParsedBody();
+
+        $dxEvolutionID  = $values['DxEvolutionID'] ?? null;
+        $name           = $values['Name'] ?? null;
+        $projectID      = $values['ProjectID'] ?? null;
+        $visit          = $values['visitLabel'] ?? null;
+        $instrumentName = $values['instrumentName'] ?? null;
+        $sourceFields   = isset($values['sourceField']) ?
+            explode(',', $values['sourceField']) : [];
+        $orderNumber    = $values['orderNumber'] ?? null;
+
+        /**
+        * VALIDATE THE FORM
+        */
+        // Validation: Form is complete
+        if (!($dxEvolutionID && $name && $visit
+            && $instrumentName && $sourceFields && $orderNumber)
+        ) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'Please fill out all fields!'
+            );
+        }
+
+        // Validation: Visit is part of Project's config
+        $projectVisits = \Utility::getVisitList(new \ProjectID($projectID));
```

Comment:
This caused a bug on CCNA when trying to set diagnosis evolution if the visits haven't yet been started, so I did a select from test_battery instead of session
```suggestion
        $projectVisits = $DB->pselectCol(
            "SELECT DISTINCT Visit_label FROM test_battery tb
            JOIN project_subproject_rel psr ON (psr.SubprojectID=tb.SubprojectID)
            WHERE psr.ProjectID=:projID
            ORDER BY Visit_label",
            ['projID' => $projectID]
        );
	```